File: leetcode/silding_window/1343/imp.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        
        int thresholdSum = k * threshold;
        int currSum = 0;
        int res = 0;

        for (int i = 0; i < k; ++i) {
            currSum += arr[i];
        }
        if (currSum >= thresholdSum) {
            ++res;
        }

        for (int i = k; i < arr.size(); ++i) {
            currSum += arr[i];
            currSum -= arr[i - k];
            if (currSum >= thresholdSum) {
                ++res;
            }
        }
        
        return res;

    }
};

File: leetcode/silding_window/1343/imp.py
from typing import List


class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:

        res = 0
        thresholdSum = k * threshold
        currSum = 0

        for i in range(k):
            currSum += arr[i]

        if currSum >= thresholdSum:
            res += 1

        for i in range(k, len(arr)):
            currSum += arr[i]
            currSum -= arr[i - k]
            
            if currSum >= thresholdSum:
                res += 1

        return res
